libs/wallet/src/connectors/rest-connector.ts

```diff
@@ -87,16 +87,22 @@ export class RestConnector implements VegaConnector {
 
   async sendTx(body: TransactionSubmission) {
     try {
-      return await this.service.commandSyncPost(body);
+      const res = await this.service.commandSyncPost(body);
+      return res;
     } catch (err) {
       return this.handleSendTxError(err);
     }
   }
 
   private handleSendTxError(err: unknown) {
-    if (typeof err === 'object' && err && 'body' in err) {
+    if (typeof err === 'object' && err && 'body' in err && 'code' in err) {
+      // @ts-ignore Not sure why TS can't infer that 'body' and 'code' do indeed exist on object
```

Comment:
comment (non-blocking-smartassness): yeah, this is pretty annoying in JS, you can pretty much throw anything, doesn't have to be an Error instance, that's why ts enforces you to treat it as 'unknown' or 'any'. another option here could be to do `if (err instanceof Error) {...}`

Reply:
I've added a type guard to tidy this up. I can't do instanceof Error here because the api client throws a plain object

```
interface ServiceError {
  code: number;
  body: string | undefined;
  headers: object;
}

export const isServiceError = (err: unknown): err is ServiceError => {
  // Some responses don't contain body object
  if (typeof err === 'object' && err !== null && 'code' in err) {
    return true;
  }
  return false;
};
```